src/bindings/python/openalpr/openalpr.py

```diff
@@ -157,6 +168,23 @@ def recognize_array(self, byte_array):
         self._free_json_mem_func(ctypes.c_void_p(ptr))
         return response_obj
 
+    def recognize_image(self, img):
+        """
+        This causes OpenALPR to attempt to recognize an image passed in as a numpy array.
+
+        :param img: This should be a 2d or 3d numpy array (gray scale or rgb image)
+        :return: An OpenALPR analysis in the form of a response dictionary
+        """
+        img = img.astype(np.ubyte)
+        # deal with gray scale and rgb images
+        dims = 3 if len(img.shape) == 3 else 1
```

Comment:
i should probably have call it depth instead of dims. because gray scale image depth is 1 but dim is 2.
I made the code regarding the depth not the dims of the image. So if used dims i need to change the c++ code